### PR TITLE
Navbar

### LINKED ISSUE #104: navbar
- [x] dark mode icon fix: bg color : midnightblue / slider: background-color: #3B82F6; / slider:checked: background-color: #607D8B;
- [x] tooltips for all tabs
- [x] fix dropdown icon
- [x]  fix small num on each tab 
- [x] search bar extend button to extend on small screens

### PR DESCRIPTION
**Pull request by mohammadalnajar**
# Description

- navbar mobile responsive 
- extend search bar on small screens onClick
- dark mode icon fixed (size and transitions)
- modals created for messenger, notifications, and searching
- added tooltips for all icons

Fixes #104 

## Type of change

- [x] New feature (non-breaking change which adds functionality)

# Checklist:

- [x] My code follows the style guidelines of this project

### CODE REVIEW

**Review by Hani-AS — COMMENTED** [review]
Left some comments for you

File: client/src/hooks/useToggleOpen.js [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+  const toggle = () => {
+    setIsOpen(!isOpen);
+  };
+
+  if (ref) {
+    useOnClickOutside(ref, () => {
+      setIsOpen(false);
+    });
+  }
+  return { isOpen, toggle };
```
[the remark this answers is not in the document]
Comment (mohammadalnajar) [reply client/src/hooks/useToggleOpen.js:15]: yes u r right man , that is easier 😁 will  fix it

[the remark this answers is not in the document]
Comment (mohammadalnajar) [reply client/src/hooks/useToggleOpen.js:15]: Done

File: client/src/hooks/useToggleOpen.js [hunk 2]
```diff
@@ -0,0 +1,18 @@
+import { useState } from 'react';
+import useOnClickOutside from './useOnClickOutside';
+
+const useToggleOpen = ({ initialState, ref = false }) => {
```
[the remark this answers is not in the document]
Comment (mohammadalnajar) [reply client/src/hooks/useToggleOpen.js:4]: Done

File: client/src/components/navbar/components/tabsMenu/Tab.jsx [hunk 3]
```diff
@@ -1,8 +1,14 @@
...
+import capitalize from '../../../../utils/helpers';
+import useToggleOpen from '../../../../hooks/useToggleOpen';
 
 const Tab = ({ selectedState, title, dynamicNum = false }) => {
   const { setSelectedTab, selectedColor } = selectedState;
+  const { isOpen: isOpenTooltip, toggle: toggleTooltip } = useToggleOpen({
+    initialState: false,
+  });
+
   const onClick = () => {
```
[the remark this answers is not in the document]
Comment (mohammadalnajar) [reply client/src/components/navbar/components/tabsMenu/Tab.jsx:12]: yeah u are right man, changed 👍

**Pull request merged**